src/main/java/mong/Mong.java

```diff
@@ -0,0 +1,358 @@
+            String line = scanner.nextLine();
+            loadLineContents(line);
+        }
+    }
+
+    /**
+     * Marks item in index as completed.
+     */
+
+    public static void mark(String input) {
+        // the itemIndex is -1 than the input from the user
+        int itemIndex = Integer.parseInt(input.split(" ")[1]) - 1;
+        if (itemIndex < 0 || itemIndex >= list.size()) {
+            throw new NullPointerException();
+        }
+        try {
+            list.get(itemIndex).setCompleted(true);
+            System.out.println("Mong >_<!!I have marked it as completed:");
+            System.out.println(list.get(itemIndex));
+        } catch (NullPointerException | IndexOutOfBoundsException e) {
+            System.out.println("Mong?!@ Item not in list.");
+        }
+    }
+
+    /**
+     * Marks completed item in index as incompleted.
+     */
+    
+    public static void unmark(String input) {
+        // the itemIndex is -1 than the input from the user
+        int itemIndex = Integer.parseInt(input.split(" ")[1]) - 1;
+        if (itemIndex < 0 || itemIndex >= list.size()) {
+            throw new NullPointerException();
+        }
+        try {
+            list.get(itemIndex).setCompleted(false);
+            System.out.println("Mong-mong :<! The task has been unmarked:");
+            System.out.println(list.get(itemIndex));
+        } catch (NullPointerException | IndexOutOfBoundsException e) {
+            System.out.println("Mong?!@ Item not in list.");
+        }
+    }
+    /**
+     * Prints a horizontal line with width of 50 characters.
+     */
+    public static void printHorizontalLine() {
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    /**
+     * Prints out an indexed list of commands given to Mong starting from 1.
+     */
+    
+    public static void printIndexedList(String input) throws IllegalTaskFormatException {
+        if (!input.contentEquals("list")) {
+            throw new IllegalTaskFormatException("Task format is incorrect!");
+        }
+        for (int i = 0; i < list.size(); i++) {
+            System.out.println(i + 1 + ". " + list.get(i));
+        }
+        System.out.println("MONG my god! You have " + list.size() + " task(s) in the list!");
+    }
+
+    /**
+     * Add a Task of type Deadline from txt file.
+     * A deadline consists of the task description and deadline in String format.
+     */
+    public static void addDeadline(String isCompleted, String description, String by) {
+        list.add(new Deadline(description, by));
+        if (isCompleted.contentEquals("0")) {
+            list.get(list.size() - 1).setCompleted(false);
+        } else if (isCompleted.contentEquals("1")) {
+            list.get(list.size() - 1).setCompleted(true);
+        }
+    }
+
+    /**
+     * Add a Task of type Deadline.
+     * A deadline consists of the task description and deadline in String format.
+     */
+    public static void addDeadline(String input) {
```

Comment:
The methods addTodo, addDeadline, and addEvent have duplicated logic. Refactor these methods to share common code for validation and task creation to avoid duplication.